grpc-kotlin-gen/src/main/resources/KtStub.mustache

```diff
@@ -105,7 +105,7 @@ object {{className}} {
             request: {{inputType}},
             responseObserver: StreamObserver<{{outputType}}>
         ) {
-            launch {
+            launch (ContextCoroutineContextElement()) {
```

Comment:
`launch` is an extension function on `CoroutineScope` which this class implements. You could add the context element to the user supplied `CoroutineContext` and have it be used by all lunched coroutines within the class.

Reply:
Yeah, I though about pursuing that option, but figured that propagating the context should be the default behaviour. Values attached to the context a quite useful and users will expect them to be available. Maybe a better approach would be to include this new context element in the default coroutineContext specified on the constructor.

My preference is to have the context propagated all the time, without the user having to think about it (even if the user specifies their own coroutineContext).

Reply:
Yes I'd prefer that too. What I meant was that it should be added to whatever context the user passes in to the constructor so it's always present throughout the class, instead of only being added to each `launch` block.

```kotlin
abstract class GreeterImplBase(
    val _coroutineContext: CoroutineContext = Dispatchers.Default,
    val sendChannelCapacity: Int = KtChannel.UNLIMITED
) : BindableService, CoroutineScope {

    override val coroutineContext: CoroutineContext
        get() = _coroutineContext + ContextCoroutineContextElement()

// ...
```

Reply:
Ok strike that. I realized this won't work because the `Context.current()` call would happen when the service instance is created and not on each call. So either we have to get the current context differently (can it be done in `updateThreadContext`?), or we have to add the context element in each `launch` block as you've done.

Reply:
Strike that too, this is full of surprises :)

So in my example, the `ContextCoroutineContextElement()` call is done in the property getter, which in turn is called by coroutine builders like `launch`. This means that the `Context.current()` will be the one from the call scope. So that works!

I would write some tests to make sure we get the expected behavior.

Reply:
Yeah, I think I like that. I'll update the PR with that change, then add a some tests.